Mathlib/Data/Matroid/Rank/Finite.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Closure
+
+/-!
+# Finite-rank sets
+
+For `M : Matroid α` and `X : Set α`,
+`Matroid.RkFin M X`  means that every basis of `X` in `M` is finite,
+or equivalently that the restriction to `X` is `Matroid.RankFinite`.
+`RkFin` sets in a matroid are the largest class of sets for which one can do nontrivial
+integer arithmetic involving the rank function.
+
+# Implementation Details
+
+Unlike most set predicates on matroids, a set `X` with `M.RkFin X` need not satisfy `X ⊆ M.E`,
+so may contain junk elements. This seems to be what makes the definition easiest to use.
+-/
+
+variable {α : Type*} {M : Matroid α} {X Y I : Set α} {e : α}
+
+open Set
+
+namespace Matroid
+
+/-- a `RkFin` set in `M` is one whose bases are all finite. -/
+def RkFin (M : Matroid α) (X : Set α) : Prop := (M ↾ X).RankFinite
+
```

Comment:
```suggestion
/-- a `RankFin` set in `M` is one whose bases are all finite. -/
def RankFin (M : Matroid α) (X : Set α) : Prop := (M ↾ X).RankFinite
```
seems more appropriate, given how rank is now written!

Reply:
No - the new convention is that `rank`/`Rank` is referring to the rank of the whole matroid, while `rk`/`Rk` is the rank of a set within the matroid.

Reply:
Ah okay, I missed that, thanks. Would `RkFinite` be better here then? The atom `Fin` usually refers to the type `Fin` (and isn't comparable to `Fintype` since that's not written as `FinType`)